apis/python/tests/test_experiment_query.py

```diff
@@ -871,3 +882,37 @@ def add_sparse_array(coll: CollectionBase, key: str, shape: Tuple[int, int]) ->
         )
     )
     a.write(tensor)
+
+@pytest.mark.parametrize("n_obs,n_vars", [(1001, 99)])
+def test_experiment_query_uses_threadpool_from_context(soma_experiment):
+    """
+    Verify that ExperimentAxisQuery uses the threadpool from the context
+    """
+
+    from concurrent import futures
+    class SentinelThreadPoolExecutor(futures.ThreadPoolExecutor):
```

Comment:
This needs to be DRY'd and put somewhere else, but I don't see a suitable place to do it (normally, `conftest.py` would be a good candidate, but it isn't present). Any preference?

Reply:
You can avoid the need for this by using a Mock:

```python

from unittest import mock

# ...

pool = mock.Mock(wraps=futures.ThreadPoolExecutor(...))
```

then use the [mock assertion methods](https://docs.python.org/3/library/unittest.mock.html#the-mock-class):

```python
pool.submit.assert_called()
# Or use the more advanced methods, like assert_called_with, etc.
```

Reply:
Ah, didn't know about `wraps=`! This is great, thanks.